CONTRIBUTION.md

```diff
@@ -0,0 +1,267 @@
+
+## 3. Creating a Pull Request (PR)
+
+This section covers how to create your changes, and submit them for review by Ping Identity engineers 
+by using a Pull Request. A PR is a formal request to merge your changes from your forked repository into 
+the main project. The following steps will guide you on creating a well-structured PR that 
+facilitates efficient review and integration of your contributions.
+
+### 1. Create a New Branch
+   Always create a new branch to work on your changes. Avoid making changes directly on the `develop` or `master` branch.
+
+   ```bash
+   git checkout -b feature/my-new-feature
+   ```
+   
+### 2. Make Your Changes
+Implement the required changes or new features. Make sure to write clean, well-commented, and readable code. If applicable, include tests and documentation for the new functionality.
+
+### 3. Commit Your Changes
+Once you’ve made your changes, commit them with a clear and descriptive message. Note that our 
+repository requires all commits to be signed. For more information on signing commits, please refer to
+the [GitHub Docs](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)
+
+The commit message should follow this structure:
+
+```
+[TYPE] Short description of the changes
+```
+Types:
+
+* `feat`: A new feature
+* `fix`: A bug fix
+* `docs`: Documentation changes
+* `refactor`: Code refactoring (no feature change)
+* `test`: Adding or modifying tests
+
+Example:
+
+   ```bash
+   git commit -S -m "feat: add login functionality"
+   ```
+
+### 4. Push Your Changes
+After committing your changes, push them to your fork:
+
+   ```bash
+    git push origin feature/my-new-feature
+   ```
+
+### 5. Create a Pull Request
+
+1. Go to your fork on GitHub.
+
+2. Click on the "New Pull Request" button.
+
+3. Select the base repository and base branch (usually `develop`), then select the branch you just pushed.
+
+4. Fill out the PR Template
+
+   Make sure to fill out the PR template provided. The template helps us better understand your change. Typically, a PR will require the following information:
+
+   * Add a title and description for the PR. The description should include:
+     * What was changed and why.
+     * Any related issues.
+     * Any additional context if necessary, for example relevant screenshots or breaking changes. 
+   
+   Once everything looks good, submit the PR for review.
+
+### 6. PR Review and Feedback
+
+Once the PR is submitted, the team will review it. Be prepared to:
+
+* Address any feedback or requested changes.
+* Keep your branch up to date with the base branch by rebasing or merging.
+
+## 4. Additional Notes
+
+* **Testing:** Please ensure that your code is well-tested. If your changes introduce new features or bug fixes, add appropriate tests to verify the behavior.
+
+* **Documentation**: Update relevant API documentation to reflect any new features or changes to existing functionality.
```

Comment:
Maybe just remind contributors to document any changes in the public interface in the code. (We generate the API docs before we release...)